src/App.tsx

```diff
@@ -15,71 +20,86 @@ export const goods = [
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleSelectClick = (good: string) => {
+    this.setState({ selectedGood: good });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        {
+          !selectedGood
+            ? <h1 className="title">No goods selected</h1>
+            : (
+              <h1 className="title is-flex is-align-items-center">
+                {`${selectedGood} is selected`}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+                <button
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                  title="reset"
+                  onClick={this.handleResetClick}
+                />
+              </h1>
+            )
+        }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map((good) => {
+              const isSelected = selectedGood === good;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              return (
+                <tr
+                  data-cy="Good"
+                  className={
+                    cn({
+                      'has-background-success-light': isSelected,
+                    })
+                  }
+                >
+                  <td>
+                    <button
+                      data-cy={
+                        cn({ AddButton: !isSelected },
+                          { RemoveButton: isSelected })
+                      }
+                      type="button"
+                      className={
+                        cn('button', { 'is-info': isSelected })
+                      }
+                      onClick={() => {
+                        return isSelected
+                          ? this.setState({ selectedGood: '' })
```

Comment:
You can use your function here, since you have it
```suggestion
                          ? this.handleResetClick()
```